source/blog/_posts/2018-08-23-against-microservices.md

```diff
@@ -0,0 +1,136 @@
+
+## Micro What?
+
+Distributed systems are all the rage now and have been at least over the past decade. The Cloud innovation, especially in the PaaS and FaaS (= keep Mores Law alive by increasing the number of parallel processes) area, pushes software development to create finer grained executable stacks for more and more abstract machines. Technologies invented for [CORBA](https://en.wikipedia.org/wiki/Common_Object_Request_Broker_Architecture), [SOA](https://en.wikipedia.org/wiki/Service-oriented_architecture) or [REST](https://en.wikipedia.org/wiki/Representational_state_transfer) are joining forces in Microservices to rule the word.
+
+Microservices are an architecture style. They are a possible solution to key demands for large scale development. Large Software, large Userbase, large development team, large data and so on. The central promisse is that they allow for **far greater scaling** then any other style of architecture by reducing the need for **global decissions**.
+
+They usually get contrasted with the vast space of **monolithic** architecture styles => an application as a single interconnected entity of functionality.  
+
+## So what is a Microservice?
+
+If you look at the - always insightfull - [blog of Martin Fowler](https://martinfowler.com/articles/Microservices.html) he describes the style as 
+
+> [...] a particular way of designing software applications as suites of **independently deployable services** [...]
+
+How would this work? When planing a green field application one typically slices the project up front in **layers/components/milestones/building blocks**. If you plan to create Microservices you are now free to slice these building blocks into different applications. Communication is only allowed to go through an API and in many cases publish and subscribe to an event system. These Services do not share infratsructure. They are (potentially) deployed to different servers, write to their own storage and know only about necessary pears, that they themselves only accesss through a well defined API. The term *micro* comes from the idea that these services themseves should be *really small* or do *one thing only*.
+
+A single services then has to be a fully working application, that does a miniscule part of the workload of the whole application. Well known and published examples of this architecture styles are [Netflix](https://medium.com/netflix-techblog) and Amazon, or eCommerce specific [Otto Group](https://dev.otto.de/tag/self-contained-systems/) or the [Spryker Plattform](https://academy.spryker.com/).
+
+Of course the devil is in the details. But simply put Microservices reduce code coupling in favor of networking and reduce organization coordination by giving more freedom to the individual teams and developers. So let's investigate the two promises separately:
```

Comment:
`The devil lies in the details`

Reply:
hmm, google sagt `is` https://en.wikipedia.org/wiki/The_devil_is_in_the_detail

Reply:
You're right! I've mistaken it with a book I've read a couple weeks ago, sorry.